src/pages/api/graphql.tsx

```diff
@@ -62,36 +53,36 @@ const clientConnection = async () => {
   }
 }
 
-export const apolloServer = new ApolloServer({
+export const apolloServer = new ApolloServer<ServerContext>({
   typeDefs,
   resolvers,
-  dataSources: () => {
-    return {
-      keystoneAPI: new KeystoneAPI(),
-      weatherAPI: new WeatherAPI(),
-      personnelAPI: new PersonnelAPI(),
-    }
-  },
+  status400ForVariableCoercionErrors: true,
   cache: 'bounded',
   plugins: [ApolloServerPluginLandingPageDisabled()],
-  context: async ({ req, res }) => {
+})
+
+export default startServerAndCreateNextHandler(apolloServer, {
+  context: async (req, res) => {
     const session = await getSession(req, res)
+    const { cache } = apolloServer
 
     if (!session || !session.passport || !session.passport.user) {
-      throw new AuthenticationError('No user in session')
+      throw new GraphQLError('No User in session', {
```

Comment:
We now have to manually create all error messages, based on GraphQLError. You'll see a lot of these changes!